src/utils/calculated_stats.py

```diff
@@ -104,6 +147,41 @@ def auto_accuracy_by_match(self, team_number: int) -> Series:
             Queries.AUTO_GRID
         ) + auto_missed_by_match  # Adding auto missed in order to get an accurate % (2 scored + 1 missed = 33%)
         return 1 - (auto_missed_by_match / auto_cycles_by_match)
+    
+    def auto_total_attempted_charge(self, team_number: int) -> int:
+        """Calculates the total times a team attempted to charge
+
+        :param team_number: The team to determine the auto accuracy per match for.
+        :return: the total times a team attempted to charge
+        """
+        attempted_charges = self.stat_per_match(team_number, Queries.AUTO_ENGAGE_ATTEMPTED).apply(
+            lambda state: int(state == "Engage") 
+        ).sum()
+
+        successful_charges = self.auto_total_successful_charge(team_number)
+
+        return successful_charges if attempted_charges < successful_charges else attempted_charges
+        
+
+    def auto_total_successful_charge(self, team_number: int) -> int:
+        """Calculates the total times a team successfully charged
+
+        :param team_number: The team to determine the auto accuracy per match for.
+        :return: the total times a team successfully charged
+        """
+        return self.stat_per_match(team_number, Queries.AUTO_CHARGING_STATE).apply(
+            lambda state: int(state == "Engage") 
+        ).sum()
+    
+    def auto_engage_success_rate(self, team_number: int) -> float:
+        """Calculates the successrate a team has for auto engaging
+
+        :param team_number: The team to determine the auto accuracy per match for.
+        :return: value from 0 to 1 that represent the percent succesrate, rounded to 3 decimal places
+        """
+        return round(self.auto_total_successful_charge(team_number) / self.auto_total_attempted_charge(team_number), 3)
```

Comment:
If taking out the two methods above, I would replace this with their counterparts (using cumulative_stat which can be found in team-graphs).